### PR TITLE
Control batch sizes during execute

### PR DESCRIPTION
**Pull request by dmitchell**
@arsduo please accept this revision which limits the batch sizes to FB's limits

### CODE REVIEW

**Comment by dmitchell** [comment]
I'm going to assume the failure for JRuby19 loading rake is completely separate from this.

**Comment by dmitchell** [comment]
Another thing this PR fixes is recursive calls. The current master resubmits the whole batch set plus the additions if you do something like the following, but this PR removes each submitted call from `batch_calls` and thus avoids the concatenation problem:

```
  api.batch do |batch_api|
     MyObject.each { |obj|
        batch_api.get_object(obj.external_id, {fields: '..'}) { |ret|
           batch_api.put_connections(obj.external_id, '', {foo: new_value_from(ret)}) { .. }
        }
     }
  end 
```

**Comment by dmitchell** [comment]
Re the build failures on faraday going into infinite recursion, I had that problem until I reinstalled the faraday gem. idk if something changed in the gem. The recursion makes no sense from the source code.

**Comment by arsduo** [comment]
Hi @dmitchell -- thanks for the awesome pull request! Sorry about the delay in getting back to you and sorry about the some issues with the testing. I've fixed the test failures (and removed a bunch of now-dead code), which should resolve that. (Looks like you unfortunately have to fix some conflicts, sorry 😐.)

I've looked over the PR and will give it another review soon. Would you be willing to split this into two separate pull requests? It's really two different changes, one more involved than the other; I'd be happy to get the batch size limit change in quickly while reviewing the recursive changes.

**Comment by dmitchell** [comment]
Yes I'll split it. Thanks for the base fixes.

File: lib/koala/api/graph_batch_api.rb [hunk 1]
```diff
@@ -8,6 +8,11 @@ class GraphBatchAPI < API
       # inside a batch call we can do anything a regular Graph API can do
       include GraphAPIMethods
 
+      # Limits from @see https://developers.facebook.com/docs/marketing-api/batch-requests/v2.8
+      MAX_AD_SUBMISSIONS = 10
```
Comment (dmitchell) [inline lib/koala/api/graph_batch_api.rb]: From what I can tell, altho FB is hard about `MAX_CALLS`, `MAX_AD_SUBMISSIONS` may be just a recommendation.

**Comment by dmitchell** [comment]
Going to create 2 new PRs with the rebase and kill this one

**Pull request closed without merging**